google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/UploadCallable.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.transfermanager;
+
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.PackagePrivateMethodWorkarounds;
+import com.google.cloud.storage.StorageException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+
+public class UploadCallable implements Callable<UploadResult> {
+  private final ExecutorService executor;
+  private final TransferManagerConfig transferManagerConfig;
+
+  private final BlobInfo originalBlob;
+
+  private final Path sourceFile;
+
+  public UploadCallable(
+      ExecutorService executor,
+      TransferManagerConfig transferManagerConfig,
+      BlobInfo originalBlob,
+      Path sourceFile) {
+    this.executor = executor;
+    this.transferManagerConfig = transferManagerConfig;
+    this.originalBlob = originalBlob;
+    this.sourceFile = sourceFile;
+  }
+
+  public UploadResult call() throws Exception {
+    // TODO: Check for chunking
+    return uploadWithoutChunking();
+  }
+
+  private UploadResult uploadWithoutChunking() throws IOException {
```

Comment:
Once we support MPU, it would be great if we could determine how (single shot vs resumable vs MPU) to upload based on the object size.

Reply:
yes, I think this is the goal. also throughput based. If we are uploading super fast no sense in spending the overhead to split the work.